proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -1124,20 +1107,12 @@ private SymbolScopeSuper classStateSuper(JPNode classNode, String className) {
       // vendor libraries, etc.
       return null;
     }
-    PUB pub = new PUB(refSession, FileStuff.fullpath(file));
-    boolean pubIsCurrent = pub.loadTo(PUB.HEADER);
     ParseUnit pu = new ParseUnit(file, refSession);
-    pu.setPUB(pub);
     JPNode superClassTree = (JPNode) classNode.getLink(IConstants.SUPER_CLASS_TREE);
     try {
       if (superClassTree != null) {
         pu.setTopNode(superClassTree);
       } else {
-        if (!pubIsCurrent)
-          // XXX Is Error really required ???
-          throw new Error("Internal error: No tree from PUB or Proparse, for class: " + className);
-        pub.load();
-        pu.setTopNode(pub.getTree());
       }
       pu.treeParser01();
     } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Catch a list of specific exception subtypes instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2221)